dev/PushNotifications/PushNotificationsLongRunningTask/winmain.cpp

```diff
@@ -1,10 +1,111 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-#include <windows.h>
 #include "pch.h"
 
+// include notifications constants file here
+
+#include <PushNotificationsLRP_h.h>
+
+// Temporarily disable C4324 because WRL generates a false (well, irrelevant) warning
+//   'Microsoft::WRL::Details::StaticStorage<Microsoft::WRL::Details::OutOfProcModuleBase<ModuleT>::GenericReleaseNotifier<T>,Microsoft::WRL::Details::StorageInstance::OutOfProcCallbackBuffer1,ModuleT>': structure was padded due to alignment specifier
+#pragma warning(push)
+#pragma warning(disable:4324)
+#include <wrl.h>
+#pragma warning(pop)
+
+#include "platform.h"
+#include "platformfactory.h"
+#include "externs.h"
+
+using namespace Microsoft::WRL;
+
+void SignalEvent()
+{
+    GetWinMainEvent().SetEvent();
```

Comment:
Instead of making the timer global, why not make it a part of platform itself. Then you can have platform helper methods:

Platform.signalEvent
Platform.Wait
Platform.cancelEvent